core/trino-main/src/main/java/io/trino/execution/SetSessionAuthorizationTask.java

```diff
@@ -91,6 +94,18 @@ else if (userExpression instanceof StringLiteral stringLiteral) {
             accessControl.checkCanImpersonateUser(originalIdentity, user);
         }
         stateMachine.setSetAuthorizationUser(user);
+        SelectedRole selectedRole;
+        Set<String> enabledRoles = originalIdentity.getEnabledRoles();
+        if (enabledRoles.isEmpty()) {
+            selectedRole = new SelectedRole(SelectedRole.Type.NONE, Optional.empty());
+        }
+        else if (enabledRoles.size() == 1) {
+            selectedRole = new SelectedRole(SelectedRole.Type.ROLE, Optional.of(enabledRoles.iterator().next()));
+        }
+        else {
```

Comment:
I don't get that. If we have `enabledRoles.size() == 2` we say we enabled all roles?

Reply:
Yes, it is not my idea, I barely took it from `io.trino.server.SessionContext#fromSession` see https://github.com/trinodb/trino/blob/c8a173a6fe073d9fbfae36d9974a0098fb3538cf/core/trino-main/src/main/java/io/trino/server/SessionContext.java#L242